### PR TITLE
Update to Go 1.22 and update deps

### PR DESCRIPTION
**Pull request by andrewkroh**
Update to Go 1.22.

Update dependencies.

Bind to 127.0.0.1 when testing.

Add dependabot.yml config to keep GH actions updated.

Remove usage of `GO111MODULE=off`.

Update to https://github.com/updatecli/updatecli-action/releases/tag/v2.56.0.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,21 +1,18 @@
 LICENSE := ASL2-Short
 VERSION ?= local
 
-check-fmt: goimports go-licenser
-	@go-licenser -d -license ${LICENSE}
-	@goimports -l -e -local github.com/elastic . | read && echo "Code differs from gofmt's style. Run 'gofmt -w .'" 1>&2 && exit 1 || true
+GOIMPORTS := go run golang.org/x/tools/cmd/goimports@latest
+GOLICENSER := go run github.com/elastic/go-licenser@latest
```
Comment (efd6) [inline Makefile:5]: Suggest that if `@latest` are used, the makefile emits the version of the tools during the build.
```
curl -s https://proxy.golang.org/golang.org/x/tools/@latest | jq '.Version'
curl -s https://proxy.golang.org/github.com/elastic/go-licenser/@latest | jq '.Version'
```

**Review by aleksmaus — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (andrewkroh) [reply Makefile:5]: I think what you are saying is that we should not use `@latest`? Based on recent pains, I would agree. Moving toward a more hermetic build would be better despite the increased effort to periodically update versions.

Comment (efd6) [reply Makefile:5]: I would prefer not to have dynamic versions, yes.

Comment (andrewkroh) [reply Makefile:5]: I opened https://github.com/elastic/stream/pull/80 to automate this.

**Pull request merged**